### PR TITLE
Feat/content cards

### PR DESCRIPTION
**Pull request by comanzanares**
Add Content Cards functionality

### CODE REVIEW

File: android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java [hunk 1]
```diff
@@ -599,6 +616,174 @@ public void getInstallTrackingId(Callback callback) {
     reportResultWithCallback(callback, null, Appboy.getInstance(getReactApplicationContext()).getInstallTrackingId());
   }
 
+  @ReactMethod
+  public void getContentCards(final Promise promise) {
+    final IEventSubscriber<ContentCardsUpdatedEvent> subscriber = new IEventSubscriber<ContentCardsUpdatedEvent>() {
+      @Override
+      public void trigger(ContentCardsUpdatedEvent event) {
+        promise.resolve(mapContentCards(event.getAllCards()));
```
Comment (stefan-sabau-3pg) [inline android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java:645]: Shouldn't the `promise.resolve` be placed as the last method call?

File: android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java [hunk 2]
```diff
@@ -599,6 +616,174 @@ public void getInstallTrackingId(Callback callback) {
...
+    WritableArray cards = Arguments.createArray();
+    for (Card card : cardsList.toArray(new Card[0])) {
+      cards.pushMap(mapContentCard(card));
+    }
+    return cards;
+  }
+
+  private WritableMap mapContentCard(Card card) {
+    WritableMap mappedCard = Arguments.createMap();
+    mappedCard.putString("id", card.getId());
```
Comment (stefan-sabau-3pg) [inline android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java]: Ideally, these keys should live under a/some symbolic constant(s), not all of them have a single occurrence.

File: iOS/AppboyReactBridge/AppboyReactBridge/AppboyReactBridge.m [hunk 3]
```diff
@@ -341,6 +362,40 @@ - (void)feedUpdated:(NSNotification *)notification {
...
+  NSArray<ABKContentCard *> *cards = [[Appboy sharedInstance].contentCardsController getContentCards];
+  NSPredicate *predicate = [NSPredicate predicateWithFormat:@"idString == %@", idString];
+  NSArray *filteredArray = [cards filteredArrayUsingPredicate:predicate];
+
+  if ([filteredArray count]) {
+    // TODO: is it fair check? Braze should not provide cards with duplicated idString
+    return filteredArray[0];
+  }
+
+  return nil;
```
Comment (stefan-sabau-3pg) [inline iOS/AppboyReactBridge/AppboyReactBridge/AppboyReactBridge.m]: Minor: lines 389-396 can be re-written as:
```
return [[cards filteredArrayUsingPredicate:predicate] firstObject];
```

File: iOS/AppboyReactBridge/AppboyReactBridge/AppboyReactBridge.m [hunk 4]
```diff
@@ -359,6 +414,85 @@ - (ABKCardCategory)getCardCategoryForString:(NSString *)category {
...
 
+RCT_REMAP_METHOD(getContentCards, getContentCardsWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  RCTLogInfo(@"getContentCards called");
+  resolve([self getMappedContentCards]);
+}
+
+RCT_EXPORT_METHOD(logContentCardClicked:(NSString *)idString) {
+  ABKContentCard * cardToClick = [self getContentCardById:idString];
+  RCTLogInfo(@"logContentCardClicked with idString %@", idString);
+  if (cardToClick) {
```
Comment (stefan-sabau-3pg) [inline iOS/AppboyReactBridge/AppboyReactBridge/AppboyReactBridge.m:419]: Is it worth considering the case when a card with a given id is missing? Like extra handling or loging a debug warning?

**Review by stefan-sabau-3pg — APPROVED** [review]
Looking good, great stuff @comanzanares

**Review by eelmoni — APPROVED** [review]

File: iOS/AppboyReactBridge/AppboyReactBridge/AppboyReactBridge.m (hunk 4 above)
Comment (comanzanares) [reply iOS/AppboyReactBridge/AppboyReactBridge/AppboyReactBridge.m:419]: I don't think we will have cards without ID, if we encounter any kind of issue we can definitely log something

File: android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java (hunk 1 above)
Comment (comanzanares) [reply android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java:645]: I tested this and I think we should be ok. first getting the cards and *if needed* update them and then remove the subscription. do you see any potential issue here?

File: android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java (hunk 2 above)
Comment (comanzanares) [reply android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java]: updated

File: iOS/AppboyReactBridge/AppboyReactBridge/AppboyReactBridge.m (hunk 3 above)
Comment (comanzanares) [reply iOS/AppboyReactBridge/AppboyReactBridge/AppboyReactBridge.m]: updated

**Pull request merged**